src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -83,9 +79,27 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
 
     @Override
     public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
+        MatchAllQueryBuilder query = new MatchAllQueryBuilder();
+        query.fromXContent(parseContext);
+        return query.toQuery(parseContext);
```

Comment:
Yes, I thought so. 
In the alternative I posted today I only have 

```
@Override
        public Query parse(QueryParseContext parseContext) throws IOException, QueryParsingException {
            return fromXContent(parseContext).toQuery();
       }
```

https://github.com/elastic/elasticsearch/compare/master...cbuescher:proto/queryrefactoring-matchall-v2#diff-b5958c2f636285e4231fcbb0ef38ebf5R167